### PR TITLE
Enhanced chromeoptions arguments and preferences

### PR DESCRIPTION
**Pull request by vinogradoff**
## Proposed changes
Chromeoptions arguments and preferences now allow to have commas in value, when parameter is masked with double quotes. see #836 
Also they are made available as Configuration to work smoothly with Selenide 5.+ versions, with multiple browser.
Usage is documented in javadoc and tests.

## Checklist
- [x] Checkstyle and unit tests pass locally with my changes by running `gradle check chrome htmlunit` command
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: src/main/java/com/codeborne/selenide/webdriver/ChromeDriverFactory.java [hunk 1]
```diff
@@ -49,40 +51,30 @@ ChromeOptions createChromeOptions(Config config, Proxy proxy) {
...
-                    "Only so-called arguments (chromeoptions.args=<values comma separated>) " +
-                    "and preferences (chromeoptions.prefs=<comma-separated dictionary of key=value> " +
-                    "are supported for the chromeoptions at the moment");
-            break;
-        }
-      }
+  private ChromeOptions transferChromeOptionsFromSystemProperties(Config config, ChromeOptions currentChromeOptions) {
+    if (!config.chromeoptionsArgs().equals("")) {
+      // Regexp from https://stackoverflow.com/a/15739087/1110503 to handle commas in values
+      Stream<String> params = Arrays.stream(config.chromeoptionsArgs().split(",(?=([^\"]*\"[^\"]*\")*[^\"]*$)"));
```
Comment (FrameBassman) [inline src/main/java/com/codeborne/selenide/webdriver/ChromeDriverFactory.java]: really magic regexp

**Comment by vinogradoff** [comment]
Agree) that's why I linked Stackoverflow answer with some explanations

Viele Grüße

Alexei Vinogradov

**Comment by FrameBassman** [comment]
@vinogradoff, That's reason of my proposal (https://github.com/selenide/selenide/pull/836)

File: src/main/java/com/codeborne/selenide/Config.java [hunk 2]
```diff
@@ -14,6 +14,8 @@
   String browserBinary();
   String pageLoadStrategy();
   DesiredCapabilities browserCapabilities();
+  String chromeoptionsArgs();
+  String chromeoptionsPrefs();
```
Comment (asolntsev) [inline src/main/java/com/codeborne/selenide/Config.java]: I don't like adding the settings. We have too lot of them. And these are chrome-specific settings in the global `Config` class...

If somebody wants to tune Chrome settings so much, he can just implement `WebDriverProvider`, right?

**Comment by vinogradoff** [comment]
@asolntsev ?

File: src/main/java/com/codeborne/selenide/webdriver/ChromeDriverFactory.java [hunk 3]
```diff
@@ -94,6 +92,17 @@ private ChromeOptions transferChromeOptionsFromSystemProperties(ChromeOptions cu
     return prefs;
   }
 
+  /**
+   * parse parameters which can come from command-line interface
+   * @param csvString comma-separated values, quotes can be used to mask spaces and commas
+   *                  Example: 123,"foo bar","bar,foo"
+   * @return values as array, quotes are preserved
+   */
+  private String[] parseCSVhandlingQuotes(String csvString) {
```
[the remark this answers is not in the document]
Comment (vinogradoff) [reply src/main/java/com/codeborne/selenide/webdriver/ChromeDriverFactory.java]: csv is actually a common abbr for "comma-separated values" - params are csv

**Pull request merged**